### ISSUE TITLE
sophia: upsert lost data

### ISSUE THREAD

**Issue by agent-0007** [tarantool/tarantool#1401]
Tarantool version: Tarantool 1.6.7-630-geab01da

I use upsert to insert data:
box.space.email:upsert({email}, {{'!', -1, email_hash_index}, {'!', -1, fiber.time()}})

From ducumentation:
"!" -  for insertion
Because this i user "!" in upsert command.

I was hoping for this kind of work:
1. email|hash1|time1
2. email|hash1|time1|hash2|time2
..................
N. email|hash1|time1|hash2|time2....|hash_N|time_N

Result for few minutes:
(looks correct at first)
```
unix/:/var/run/tarantool/grepmaillog_sofia.control> box.space.email:select({'XXXX@bk.ru'})
—-
- - ['XXXX@bk.ru', '1330a9689116a75715ec233f5538404c', 1458754630.1612, '4e0c144ab41f0d0ba491ddc065756304',
    1458754630.8022, 'f8d4e9e9833ecf517a9f078caa589507', 1458754631.1123, 'd078c72bbdd1c1f308952f556a250fa1',
    1458754631.6449, '577765247168f66fe12b4ee84e5e878c', 1458754631.6843, '44aadde8a34fb05b6187d6b4550a3767',
    1458754633.0255, 'cbf3dbe447c45886e56d6df0883ad130', 1458754635.1923, 'cbe9494270aa749f6750022d94e283cc',
    1458754635.3907, '3c9b06cd855449ff99a0b516c58570a3', 1458754635.4149, '6eb65d1f84f17d00e80570c2668994a0',
    1458754638.2165, 'e6bfd1fbf95919c5e9f2775fc21e7ad5', 1458754638.6096, '079a7888ea1fc5aa8b66ff6b372e8d93',
    1458754638.9304]
```

suddenly my data has been lost:
```
unix/:/var/run/tarantool/grepmaillog_sofia.control> box.space.email:select({XXXX@bk.ru'})
—-
- - ['XXXX@bk.ru', '1330a9689116a75715ec233f5538404c', 1458754630.1612, '4e0c144ab41f0d0ba491ddc065756304',
    1458754630.8022]
```

This problem i can repeat. If you have interst - write me ;)

**Comment by pmwkaa** [comment]
There are been some issues which could result in such behavior in older versions of tarantool.
Could you please update to the latest 1.6.8 master?

**Comment by agent-0007** [comment]
My tarantool was installed from epel. If i run yum update it will become 1.6.8 ?
Or i need build by hands from branch master?

**Comment by kostja** [comment]
* agent-0007 <notifications@github.com> [16/03/24 12:20]:
> My tarantool was installed from epel. If i run yum update it will become 1.6.8 ?
> Or i need build by hands from branch master?

There is no branch master any more. yum update should be ok.

-- 
Konstantin Osipov, Moscow, Russia, +7 903 626 22 32
http://tarantool.org - www.twitter.com/kostja_osipov

**Comment by agent-0007** [comment]
I have upgrade my tarantool to version: Tarantool 1.6.8-586-g504e151
In new version tarantool upsert command writes garbage to tuple:

```
['XXXXXX@mail.ru', 114, 117, 114, 117, '3f9cf2f39e8c1caa25bc1265e80147b5',
    1458816126.5608, '549209b0615160fab35a3b0c82e4cb12', 1458816217.0361, 'c7945da08da79e6492f1493f98de895d',
    1458816406.7516, '7a3b4b93055eca47d164d271e112129d', 1458816452.0767, 'b3a1a8d9c9a60863981a22d40847ea08',
    1458816701.9667, '7e53b54e5c7231e5b6083bf2fe7cd194', 1458816774.0969, '1061ef16c9b36cfe60d7775f6a159ac4',
    1458817228.6487, '784db8623258cb9b75bbe06fb649583c', 1458817237.5723, '9482dfd7a77b5fcb5a73c5fd3c940abb',
    1458817238.1713]
```

later:

```
['XXXXXX@mail.ru', 114, 117, 114, 117, 114, 117, 114, 117, 114, 117, '3f9cf2f39e8c1caa25bc1265e80147b5',
    1458816126.5608, '549209b0615160fab35a3b0c82e4cb12', 1458816217.0361, 'c7945da08da79e6492f1493f98de895d',
    1458816406.7516, '7a3b4b93055eca47d164d271e112129d', 1458816452.0767, 'b3a1a8d9c9a60863981a22d40847ea08',
    1458816701.9667, '7e53b54e5c7231e5b6083bf2fe7cd194', 1458816774.0969, '1061ef16c9b36cfe60d7775f6a159ac4',
    1458817228.6487, '784db8623258cb9b75bbe06fb649583c', 1458817237.5723, '9482dfd7a77b5fcb5a73c5fd3c940abb',
    1458817238.1713]
```

This is not my data: 114, 117, 114, 117, 114, 117, 114, 117, 114, 117 - it's garbage.

**Comment by pmwkaa** [comment]
merged into 1.6 and 1.7
